app/src/main/java/com/edulectronics/tinycircuit/Views/CircuitActivity.java

```diff
@@ -241,40 +221,23 @@ public void onClickAddComponent(String text) {
         ((DrawerLayout) findViewById(R.id.activity_main)).closeDrawer(view);
     }
 
-    public ExpandableListView.OnGroupClickListener onGroupClick() {
-        return new ExpandableListView.OnGroupClickListener() {
-            @Override
-            public boolean onGroupClick(ExpandableListView parent, View v, int groupPosition, long id) {
-                if (groupPosition == 0) {
-                    LinearLayout group = (LinearLayout) v;
-                    toggleMode(group);
-                    NavigationView view = (NavigationView) findViewById(R.id.navigationview);
-                    ((DrawerLayout) findViewById(R.id.activity_main)).closeDrawer(view);
-                }
-                return false;
-            }
-        };
-    }
-
-    public ExpandableListView.OnChildClickListener onChildClick() {
-        return new ExpandableListView.OnChildClickListener() {
-            @Override
-            public boolean onChildClick(ExpandableListView parent, View v, int groupPosition, int childPosition, long id) {
-                LinearLayout group = (LinearLayout) ((LinearLayout) v).getChildAt(0);
-                View child = group.getChildAt(1);
-                String text = (String) ((TextView) child).getText();
+    public ListView.OnItemClickListener onItemClick() {
+        return new ListView.OnItemClickListener() {
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                LinearLayout group = (LinearLayout) view;
+                TextView textView = (TextView) group.findViewById(R.id.submenu);
+                String text = (String) textView.getText();
                 onClickAddComponent(text);
-                return false;
             }
         };
     }
 
-    public void toggleMode(LinearLayout linearLayout) {
+    public void toggleMode(View view) {
         isInWireMode = !isInWireMode;
         if (isInWireMode) {
-            linearLayout.setBackgroundResource(R.color.wiremode_on);
+            view.setBackgroundResource(R.color.wiremode_on);
         } else {
-            linearLayout.setBackgroundResource(R.color.wiremode_off);
+            view.setBackgroundResource(R.color.wiremode_off);
```

Comment:
Als de wiremode aanstaat dan maakt hij hem grijs om aan te geven dat hij aanstaat en anders is hij al R.color.background en er moet wel verschil zijn